
Only link HTTPD using -export-dynamic, not suexec, ab etc.

Upstream-HEAD: committed
Upstream-Status: Build system cleanup, medium risk

--- httpd-2.0.48/modules/aaa/config.m4.dynamic
+++ httpd-2.0.48/modules/aaa/config.m4
@@ -18,6 +18,4 @@
   fi
 ])
 
-APR_ADDTO(LT_LDFLAGS,-export-dynamic)
-
 APACHE_MODPATH_FINISH
--- httpd-2.0.48/modules/arch/win32/config.m4.dynamic
+++ httpd-2.0.48/modules/arch/win32/config.m4
@@ -6,6 +6,4 @@
 
 APACHE_MODULE(isapi, isapi extension support, , , no)
 
-APR_ADDTO(LT_LDFLAGS,-export-dynamic)
-
 APACHE_MODPATH_FINISH
--- httpd-2.0.48/modules/echo/config.m4.dynamic
+++ httpd-2.0.48/modules/echo/config.m4
@@ -6,6 +6,4 @@
 
 APACHE_MODULE(echo, ECHO server, , , no)
 
-APR_ADDTO(LT_LDFLAGS,-export-dynamic)
-
 APACHE_MODPATH_FINISH
--- httpd-2.0.48/modules/metadata/config.m4.dynamic
+++ httpd-2.0.48/modules/metadata/config.m4
@@ -18,6 +18,4 @@
 APACHE_MODULE(unique_id, per-request unique ids)
 APACHE_MODULE(setenvif, basing ENV vars on headers, , , yes)
 
-APR_ADDTO(LT_LDFLAGS,-export-dynamic)
-
 APACHE_MODPATH_FINISH
--- httpd-2.0.48/modules/cache/config.m4.dynamic
+++ httpd-2.0.48/modules/cache/config.m4
@@ -6,6 +6,4 @@
 
 APACHE_MODULE(file_cache, File cache, , , no)
 
-APR_ADDTO(LT_LDFLAGS,-export-dynamic)
-
 APACHE_MODPATH_FINISH
--- httpd-2.0.48/modules/mappers/config9.m4.dynamic
+++ httpd-2.0.48/modules/mappers/config9.m4
@@ -41,6 +41,10 @@
     enable_so="static"
 fi
 
+if test "x$enable_so" = "xstatic"; then
+    APR_ADDTO(HTTPD_LDFLAGS,-export-dynamic)
+fi
+
 if test "$sharedobjs" = "yes"; then
     if test $ac_cv_define_APR_HAS_DSO = "no"; then
         AC_MSG_ERROR([shared objects have been requested but cannot be built since mod_so cannot be built])
--- httpd-2.0.48/modules/filters/config.m4.dynamic
+++ httpd-2.0.48/modules/filters/config.m4
@@ -7,8 +7,6 @@
 APACHE_MODULE(ext_filter, external filter module, , , most)
 APACHE_MODULE(include, Server Side Includes, , , yes)
 
-APR_ADDTO(LT_LDFLAGS,-export-dynamic)
-
 APACHE_MODULE(deflate, Deflate transfer encoding support, , , no, [
   AC_ARG_WITH(z, APACHE_HELP_STRING(--with-z=DIR,use a specific zlib library),
   [
--- httpd-2.0.48/modules/generators/config5.m4.dynamic
+++ httpd-2.0.48/modules/generators/config5.m4
@@ -11,8 +11,6 @@
 APACHE_MODULE(suexec, set uid and gid for spawned processes, , , no, [
               other_targets=suexec ] )
 
-APR_ADDTO(LT_LDFLAGS,-export-dynamic)
-
 if test "$apache_cv_mpm" = "worker" -o "$apache_cv_mpm" = "perchild"; then
 # if we are using a threaded MPM, we will get better performance with
 # mod_cgid, so make it the default.
